### ISSUE TITLE
homebrew: mu 1.10.6 build issue

### ISSUE THREAD

**Issue by chenrui333** [djcb/mu#2531]
With this [guile extension patch](https://github.com/djcb/mu/pull/2522), it currently breaks [the homebrew build](https://github.com/Homebrew/homebrew-core/pull/138842)

```
PermissionError: [Errno 1] Operation not permitted: '/opt/homebrew/Cellar/guile/3.0.9/lib/guile/3.0/extensions/libguile-mu.dylib'
```

Let me know if that makes sense. Thanks!

**Comment by djcb** [comment]
Hmm, this change is there so we install things it the correct directory (as per Debian policy), and it does seem that it's the correct place... any reason why it doesn't work for homebrew?

**Comment by djcb** [comment]
Or put a different way, are there other modules installing guile modules supported by homebrew? How do they do it?

**Comment by chenrui333** [comment]
permission issue with homebrew, so it works with debian?

**Comment by chenrui333** [comment]
let me explore a bit on the homebrew side later.

**Comment by djcb** [comment]
I suspect homebrew wants packages to install in some package-specific directory. With the change we now use the equivalent of:
```
 pkg-config guile-3.0 --variable extensiondir
```
which points to somewhere under _guile's_ package-specific dir, rather than mu's, i.e. the `/opt/homebrew/Cellar/guile/3.0.9/lib/guile/3.0/extensions/`.

So I guess adding an option to the meson setup so you can pass:
```
meson setup -Dguile-extension-dir=<some-path> build  
```
could work?

**Comment by chenrui333** [comment]
> meson setup -Dguile-extension-dir=<some-path> build

checking on this.

**Comment by djcb** [comment]
Note - that's not implemented yet. But would something like that work for Homebrew? If so, I can add it.

**Comment by chenrui333** [comment]
that might be better, right now, the extension installation is mostly permission issue (installing to guile package directory rather than mu package)

**Comment by chenrui333** [comment]
but you might also need to instruct me what would be the preferred directory setup inside mu package as well :)

**Comment by djcb** [comment]
I've pushed https://github.com/djcb/mu/commit/52e3a4766e89e066b58bb871f45e4c15dbf40f9f for this; that will let you do
 ```
$  meson setup -Dguile-extension-dir=<some-path> build  
```
by default, `mu` uses the `extension-dir` that the guile-3.0 package defines, which is  `/usr/local/lib/guile/3.0/extensions` (it doesn't use `{prefix}`).

Anyway, guessing Brew conventions, I guess it should use something like `/opt/homebrew/Cellar/mu/<version>/lib/guile/3.0/extensions/`? 

If the mech works for you, I can back-port it to 1.10.x.

**Comment by djcb** [comment]
The guile-extension-dir customization is now available in mu 1.10.7; closing this.
